app/models/repeatable_step.rb

```diff
@@ -0,0 +1,78 @@
+class RepeatableStep < Step
+  include ActionView::Helpers::SanitizeHelper
+  include ActionView::Helpers::TagHelper
+  include ActionView::Helpers::OutputSafetyHelper
+
+  attr_accessor :answer_id, :questions
+
+  def initialize(...)
+    super
+    @questions = []
+  end
+
+  def repeatable?
+    true
+  end
+
+  def parent_question
+    Step.instance_method(:question).bind(self).call
```

Comment:
This needs more explanation in the code

Reply:
This does feel a little hacky. Can you think of a better name for it? It's kind of the prototype or base question for this Step but I'm not sure either of those make it clearer.

Would changing the constructor for RepeatableStep make it clearer?

```ruby
def initialize(question:, page:, form_id:, form_slug:, next_page_slug:, page_slug:)
super(question: question_prototype, page:, form_id:, form_slug:, next_page_slug:, page_slug:)
@questions = []
@question_protoype = question
```

Reply:
Is it better to do `.dup` on a question instance rather than `.new` on a class? Personally I'd suggest having an instance variable called `@question_class` and call `@question_class.new` when we want to create a new question.

Also, it just occurred to me, because we inherit from `Step` there should be an instance variable `@question` that we can use instead of calling the superclass?

```suggestion
    @question
```

We could use that in the constructor as well

```
def initialize(...)
  super
  @questions = []
  @question_protoype = @question # or @question_class = @question.class
end
```

Reply:
Thats a good idea, I'll give the instance variable a go.

I started off using question.class.new. A new question doesn't include the values taken from the page to configure the question. Question text, hint, guidance etc all get configured when the question is created.